### ISSUE TITLE
Port 9200 is used as the default, regardless of the schema

### ISSUE THREAD

**Issue by greatvovan** [elastic/elasticsearch-py#1385]
**Elasticsearch version**: not important

**`elasticsearch-py` version**: 7.9.1

**Description of the problem including expected versus actual behavior**:

The inference of default port must rely on the URL scheme ("http", "https", "amqp", etc) and not on the default port for a specific product. I have ElasticSearch exposed on port 80 (and it is not under my control) but cannot access it using URL `http://host.domain` because the library tries to connect to port 9200. An easy workaround is to specify port in the URL (`http://host.domain:80`) but this is misleading.

It is acceptable though to use port 9200 if user specifies host hame only (not a URL).

**Comment by sethmlarson** [comment]
I agree with you that the host configuration via URL shouldn't be applying the default port in the absence of one in the URL. However this would be a breaking change so likely won't come until v8. For now I recommend configuring all parts of the connection, especially those that aren't defaults.

**Comment by sethmlarson** [comment]
Thanks for reporting this, it'll be changing as a part of v8.0.
